### PR TITLE
Add versioning of local scores

### PR DESCRIPTION
**Pull request by peppy**
For any potential future usage

Minimal implementation as to not spend too much time, just want to get this in place in case we need it.

### CODE REVIEW

File: osu.Game/Scoring/ScoreInfo.cs [hunk 1]
```diff
@@ -46,6 +46,12 @@ public class ScoreInfo : RealmObject, IHasGuidPrimaryKey, IHasRealmFiles, ISoftD
         /// </remarks>
         public BeatmapInfo? BeatmapInfo { get; set; }
 
+        /// <summary>
+        /// The version of the client this score was set using.
+        /// Sourced from <see cref="OsuGameBase.Version"/> at the point of score submission.
+        /// </summary>
+        public string Version { get; set; } = string.Empty;
```
Comment (bdach) [inline osu.Game/Scoring/ScoreInfo.cs:53]: I guess probably the only question I'd have here is - do we wanna put this in replays too (and decode it from there on import)? We can probably put this in `LegacyReplaySoloScoreInfo` with practically zero effort.

Comment (peppy) [reply osu.Game/Scoring/ScoreInfo.cs:53]: Oh yeah. I didn't realise this was so low effort, if so then let's do it.

Comment (bdach) [reply osu.Game/Scoring/ScoreInfo.cs]: Have done so in https://github.com/ppy/osu/pull/25998/commits/2baf579f7ce62429e83aa4781ec3d55320b92d09 (with test coverage in https://github.com/ppy/osu/pull/25998/commits/5ff95db02c2af9ce6e7b1d707ee88440e3df9e6e).

**Comment by bdach** [comment]
As discussed on stream I've also renamed the property to `ClientVersion` in https://github.com/ppy/osu/pull/25998/commits/81bbdccee788daac7c7865859673eed1521191e8 because I feel we will have too many versions of everything otherwise and invariably get confused.

@peppy would appreciate double-check of changes. Importantly, if you're going to run this, remove `client_40.realm*` from the development directory, or else it will probably yell that the db is corrupt. This is because I did the rename without bumping schema version again.

**Review by bdach — APPROVED** [review]

**Pull request merged**